app/Http/Controllers/API/V1/ReparationRequestMaterialController.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers\API\V1;
+
+use App\Http\Controllers\Controller;
+use App\Models\ReparationRequestMaterial;
+use Illuminate\Http\Request;
+use Illuminate\Pagination\LengthAwarePaginator;
+use Spatie\QueryBuilder\AllowedFilter;
+use Spatie\QueryBuilder\AllowedInclude;
+use Spatie\QueryBuilder\QueryBuilder;
+
+class ReparationRequestMaterialController extends Controller
+{
+    public function index(Request $request): LengthAwarePaginator
+    {
+        $materialRequest = QueryBuilder::for(ReparationRequestMaterial::class)
+            ->allowedFilters([
+                AllowedFilter::partial('uuid'),
+                AllowedFilter::partial('name'),
+                AllowedFilter::partial('description'),
+                AllowedFilter::exact('is_mandatory'),
+            ])
+            ->allowedSorts([
+                'created_at',
+                'updated_at',
+            ])
+            ->allowedIncludes([
+                AllowedInclude::relationship('reparationRequest'),
+            ])
+            ->jsonPaginate();
+
+        return $materialRequest;
```

Comment:
We agreed on using specific resource classed right? Then this should be picked up.